File: packages/insomnia-httpsnippet/test/fixtures/output/ocaml/cohttp/multipart-file.ml
open Cohttp_lwt_unix
open Cohttp
open Lwt

let uri = Uri.of_string "http://mockbin.com/har" in
let headers = Header.add (Header.init ()) "content-type" "multipart/form-data; boundary=---011000010111000001101001" in
let body = Cohttp_lwt_body.of_string "-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"foo\"; filename=\"hello.txt\"\r\nContent-Type: text/plain\r\n\r\n\r\n-----011000010111000001101001--\r\n" in

Client.call ~headers ~body `POST uri
>>= fun (res, body_stream) ->
  (* Do stuff with the result *)

File: packages/insomnia-httpsnippet/test/fixtures/output/node/unirest/multipart-file.js
var fs = require('fs');
var unirest = require('unirest');

var req = unirest('POST', 'http://mockbin.com/har');

req.headers({
  'content-type': 'multipart/form-data; boundary=---011000010111000001101001'
});

req.multipart([
  {
    body: fs.createReadStream('test/fixtures/files/hello.txt'),
    'content-type': 'text/plain'
  }
]);

req.end(function(res) {
  if (res.error) throw new Error(res.error);

  console.log(res.body);
});
